pkg/proc/eval.go

```diff
@@ -3005,6 +3001,148 @@ func (v *Variable) reslice(low int64, high int64, trustLen bool) (*Variable, err
 	return r, nil
 }
 
+func (v *Variable) findStructMemberOrMethod(name string) (*Variable, error) {
+	if v.Unreadable != nil {
+		return v.clone(), nil
+	}
+	vname := v.Name
+	if v.Name == "" {
+		vname = v.DwarfType.String()
+	}
+	if v.loaded && (v.Flags&VariableFakeAddress) != 0 {
+		for i := range v.Children {
+			if v.Children[i].Name == name {
+				return &v.Children[i], nil
+			}
+		}
+		return nil, fmt.Errorf("%s has no member %s", vname, name)
+	}
+	closure := false
+	switch v.Kind {
+	case reflect.Chan:
+		v = v.clone()
+		v.RealType = godwarf.ResolveTypedef(&(v.RealType.(*godwarf.ChanType).TypedefType))
+	case reflect.Interface:
+		v.loadInterface(0, false, LoadConfig{})
+		if len(v.Children) > 0 {
+			v = &v.Children[0]
+		}
+	case reflect.Func:
+		fn := v.bi.PCToFunc(v.Base)
+		v.loadFunctionPtr(0, LoadConfig{MaxVariableRecurse: -1})
+		if v.Unreadable != nil {
+			cst := fn.extra(v.bi).closureStructType
+			if cst == nil || cst.ByteSize == 0 {
+				// Not a closure, normal function
+				if _, ok := v.bi.PackageMap[vname]; ok {
+					return nil, fmt.Errorf("package %s has no function %s", vname, name)
+				}
+				return nil, fmt.Errorf("%s has no member %s", vname, name)
+			}
+			return nil, v.Unreadable
+		}
+		if v.closureAddr != 0 {
+			fn = v.bi.PCToFunc(v.Base)
+			if fn != nil {
+				cst := fn.extra(v.bi).closureStructType
+				v = v.newVariable(v.Name, v.closureAddr, cst, v.mem)
+				closure = true
+			}
+		}
+	}
+
+	queue := []*Variable{v}
+	seen := map[string]struct{}{}
+	first := true
+
+	for len(queue) > 0 {
+		v := queue[0]
+		queue = append(queue[:0], queue[1:]...)
+		if _, isseen := seen[v.RealType.String()]; isseen {
+			continue
+		}
+		seen[v.RealType.String()] = struct{}{}
+
+		typ := v.DwarfType
+		ptyp, isptr := typ.(*godwarf.PtrType)
+		if isptr {
+			typ = ptyp.Type
+		}
+
+		var pkg, receiver string
+		typePath := typ.Common().Name
+		dot := strings.LastIndex(typePath, ".")
+		if dot >= 0 {
+			pkg, receiver = typePath[:dot], typePath[dot+1:]
+		}
+
+		if len(pkg) > 0 && len(receiver) > 0 {
+			rv, err := lookupMethod(v, isptr, pkg, receiver, name)
+			if err != nil {
+				return nil, err
+			} else if rv != nil {
+				return rv, nil
+			}
+		}
+
+		// queue embedded fields for search
+		structVar := v.maybeDereference()
+		structVar.Name = v.Name
+		if structVar.Unreadable != nil {
+			return structVar, nil
+		}
+		switch t := structVar.RealType.(type) {
+		case *godwarf.StructType:
+			for _, field := range t.Field {
+				if field.Name == name {
+					return structVar.toField(field)
+				}
+				if len(queue) == 0 && field.Name == "&"+name && closure {
+					f, err := structVar.toField(field)
+					if err != nil {
+						return nil, err
+					}
+					return f.maybeDereference(), nil
+				}
+				isEmbeddedStructMember :=
```

Comment:
Delete all of this and just use field.Embedded. It's been there since 2017, it's legacy code we don't need anymore.